### PR TITLE
feat(ci): automatic changelog

### PR DESCRIPTION
**Pull request by nuxencs**
# Pull Request

## Purpose

Automatically generate a changelog from conventional commit messages either weekly or manually.

## Approach

Fetch the last time the `updates.txt` was edited, check all the commits since that date using `git log` and then filter out bot commits and commits not following conventional commits.

## Open Questions and Pre-Merge TODOs

- [ ] filter out more stuff?

## Requirements

- [x] These changes meet the standards for [contributing](https://github.com/TRaSH-Guides/Guides/blob/master/CONTRIBUTING.md).
- [x] I have read the [code of conduct](https://github.com/TRaSH-Guides/Guides/blob/master/.github/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: .github/scripts/update-changelog.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+log_file="$1"
+output_file="docs/updates.txt"
+current_time=$(date +"%F %H:%M")
```
Comment (bakerboy448) [inline .github/scripts/update-changelog.sh:5]: Confirm this will be in UTC/Z timezone

Comment (nuxencs) [reply .github/scripts/update-changelog.sh:5]: it is UTC yes

File: .github/workflows/automatic-changelog.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      - name: Commit and push if it's not up to date
+        if: env.NEW_COMMITS == 'true'
+        env:
+          GITHUB_TOKEN: ${{ secrets.PAT }}
+        run: |
+          git config --global user.email "actions@github.com"
+          git config --global user.name "GitHub Actions"
+          git diff
+          git commit -am "chore(changelog): Update updates.txt" || exit 0
+          git push https://x-access-token:${{ secrets.PAT }}@github.com/${{ github.repository }}.git HEAD:master --force
```
Comment (bakerboy448) [inline .github/workflows/automatic-changelog.yml:55]: Don't force push!! 0 reason to do this

If needed it should be force push with lease...but if changelog updates need to be force pushed we have a serious repo issue

**Review by bakerboy448 — CHANGES_REQUESTED** [review]

**Comment by bakerboy448** [comment]
Conceptually looks good to me

got an example/local output to verify?

**Comment by nuxencs** [comment]
The top 4 runs here
https://github.com/nuxencs/Guides/actions/workflows/automatic-changelog.yml

**Comment by nemchik** [comment]
> Conceptually looks good to me
> 
> got an example/local output to verify?

https://github.com/nuxencs/Guides/blob/efeb250e47798c1597a05733e956d14e7c3af814/docs/updates.txt is probably what you're asking for
https://github.com/nuxencs/Guides/commit/efeb250e47798c1597a05733e956d14e7c3af814 and an example commit

**Review by bakerboy448 — APPROVED** [review]

**Pull request merged**